TrackingProduction/Fun4All_raw_hit_KFP.C

```diff
@@ -0,0 +1,713 @@
+
+    auto cleaner = new PHTrackCleaner();
+    cleaner->Verbosity(0);
+    cleaner->set_pp_mode(TRACKING::pp_mode);
+    se->registerSubsystem(cleaner);
+
+    if (G4TRACKING::SC_CALIBMODE)
+    {
+      /*
+      * in calibration mode, calculate residuals between TPC and fitted tracks,
+      * store in dedicated structure for distortion correction
+      */
+      auto residuals = new PHTpcResiduals;
+      const TString tpc_residoutfile = theOutfile + "_PhTpcResiduals.root";
+      residuals->setOutputfile(tpc_residoutfile.Data());
+      residuals->setUseMicromegas(G4TRACKING::SC_USE_MICROMEGAS);
+
+      // matches Tony's analysis
+      residuals->setMinPt( 0.2 );
+
+      // reconstructed distortion grid size (phi, r, z)
+      residuals->setGridDimensions(36, 48, 80);
+      se->registerSubsystem(residuals);
+    }
+
+  }
+
+  auto finder = new PHSimpleVertexFinder;
+  finder->Verbosity(0);
+  finder->setDcaCut(0.5);
+  finder->setTrackPtCut(-99999.);
+  finder->setBeamLineCut(1);
+  finder->setTrackQualityCut(1000000000);
+  finder->setNmvtxRequired(3);
+  finder->setOutlierPairCut(0.1);
+  se->registerSubsystem(finder);
+
+//   TString residoutfile = theOutfile + "_resid.root";
+//   std::string residstring(residoutfile.Data());
+
+//   auto resid = new TrackResiduals("TrackResiduals");
+//   resid->outfileName(residstring);
+//   resid->alignment(false);
+
+//   // adjust track map name
+//   if(G4TRACKING::SC_CALIBMODE && !G4TRACKING::convert_seeds_to_svtxtracks)
+//   {
+//     resid->trackmapName("SvtxSiliconMMTrackMap");
+//     if( G4TRACKING::SC_USE_MICROMEGAS )
+//     { resid->set_doMicromegasOnly(true); }
+//   }
+
+//   resid->clusterTree();
+//   resid->hitTree();
+//   resid->convertSeeds(G4TRACKING::convert_seeds_to_svtxtracks);
+
+//   resid->Verbosity(0);
+//   se->registerSubsystem(resid);
+
+  //auto ntuplizer = new TrkrNtuplizer("TrkrNtuplizer");
+  //se->registerSubsystem(ntuplizer);
+  /*
+  TString outputFile = "/sphenix/user/aopatton/TrackSeedOutput/output/RawHitFileLists_nSkip_"+std::to_string(nSkip)+ "-" + std::to_string(runnumber) + "-" + std::to_string(segment) + ".root";
+  std::string theOutputFile = outputFile.Data();
+  Fun4AllOutputManager *out = new Fun4AllDstOutputManager("out", theOutputFile.c_str());
+  se->registerOutputManager(out);
+  */
+  if (Enable::QA)
+  {
+    se->registerSubsystem(new TpcRawHitQA);
+    se->registerSubsystem(new MvtxClusterQA);
+    se->registerSubsystem(new InttClusterQA);
+    se->registerSubsystem(new TpcClusterQA);
+    se->registerSubsystem(new MicromegasClusterQA);
+    auto tpcqa = new TpcSeedsQA;
+    tpcqa->setTrackMapName("TpcSvtxTrackMap");
+    tpcqa->setVertexMapName("TpcSvtxVertexMap");
+    tpcqa->setSegment(rc->get_IntFlag("RUNSEGMENT"));
+    se->registerSubsystem(tpcqa);
+
+  }
+
+  //KFParticle dependancy
+     Global_Reco();
+
+  //KFParticle setup
+
+  KFParticle_sPHENIX *kfparticle = new KFParticle_sPHENIX("pipi_reco");
```

Comment:
Can we add a flag to turn on/off the KFP stuff, as initially we will need to do some tracking studies and many won't need KFP to be run